File: extract_excel.py
import pandas as pd
import json

file_path = r'c:\Users\EmuPC\Desktop\Proyectos\pros dc\REF\Base de datos DC.xlsx'
try:
    xl = pd.ExcelFile(file_path)
    sheet_data = {}
    for sheet_name in xl.sheet_names:
        df = xl.parse(sheet_name, nrows=5)
        sheet_data[sheet_name] = {
            "columns": df.columns.tolist(),
            "sample_rows": df.values.tolist()
        }
    print(json.dumps(sheet_data, indent=2))
except Exception as e:
    print(f"Error: {e}")

File: extract_excel.js
import * as XLSX from 'xlsx';
import * as fs from 'fs';

const filePath = 'REF/Base de datos DC.xlsx';

try {
    const fileBuffer = fs.readFileSync(filePath);
    const workbook = XLSX.read(fileBuffer, { type: 'buffer' });
    
    const result = {};
    workbook.SheetNames.forEach(sheetName => {
        const worksheet = workbook.Sheets[sheetName];
        const jsonData = XLSX.utils.sheet_to_json(worksheet, { header: 'A', range: 0 });
        result[sheetName] = {
            sampleRows: jsonData.slice(0, 10)
        };
    });
    
    console.log(JSON.stringify(result, null, 2));
} catch (error) {
    console.error('Error:', error);
}
